tox.ini

```diff
@@ -0,0 +1,56 @@
+[tox]
+envlist =
+    check-{style,build}
+    test{,-warnings,-cov}-xdist
+
+# tox environments are constructed with so-called 'factors' (or terms)
+# separated by hyphens, e.g. test-devdeps-cov. Lines below starting with factor:
+# will only take effect if that factor is included in the environment name. To
+# see a list of example environments that can be run, along with a description,
+# run:
+#
+#     tox -l -v
+#
+
+[testenv:check-style]
+description = check code style, e.g. with flake8
+skip_install = true
+deps =
+    ruff
+commands =
+    ruff . {posargs}
+
+[testenv:check-build]
+description = check build sdist/wheel and a strict twine check for metadata
+skip_install = true
+deps =
+    build
+    twine>=3.3
+commands =
+    python -m build .
+    twine check --strict dist/*
+
+[testenv]
+description =
+    run tests
+    devdeps: with the latest developer version of key dependencies
+    warnings: treating warnings as errors
+    cov: with coverage
+    xdist: using parallel processing
+extras =
+    test
+deps =
+    cov: pytest-cov
+    xdist: pytest-xdist
+commands_pre =
+    devdeps: pip install -r requirements-dev.txt -U --upgrade-strategy eager
```

Comment:
The point of using tox.ini is that you can get rid of the `requirements-dev.txt` file altogether.

Example: https://github.com/spacetelescope/synphot_refactor/blob/master/tox.ini